### ISSUE TITLE
Wayland Purple Screen

### ISSUE THREAD

**Issue by Tom-Finke** [linux-surface/linux-surface#648]
I installed Ubuntu on my Surface Pro 5. I am still in the process of optimizing everything and installing utilities. In the [Installation and Setup Guide](https://github.com/linux-surface/linux-surface/wiki/Installation-and-Setup) it says at the very bottom "You should make sure that you are running a Wayland desktop session, instead of an X11 / Xorg session." So I logged out, chose "Ubuntu with Wayland" on the login screen and logged in. Then I just got a purple screen with the cursor in the bottom right corner.

I found an issue on askubuntu with might relate to this, 

- https://askubuntu.com/questions/1028665/ubuntu-18-04-stuck-at-purple-screen-after-login
- https://askubuntu.com/questions/1150462/ubuntu-18-04-boot-freeze-purple-screen-hangs

but they either installed current graphics drivers to fix it, which I have and didnt work,  or just disabled wayland, which of course isnt an option for me because this occured because I wanted to switch to Wayland in the first place.

So: Is the advantage of Wayland over xorg regarding Touch Input actually that great, so is it worth putting in the effort?
And: Anybody got any idea what to do about this? 

Thank you in advance

# Environment

    Model: Surface Pro 5 (2017)
    Kernel version: 5.15.3-surface
    Distribution: Ubuntu 20.04.3 LTS with Gnome 3.36.8

**Comment by Amondale** [comment]
Hey @Tom-Finke,
Haven't had that experience yet (didn't mention the Ubuntu version): Purple Screen of Purgatory. However, sounds like a 20 or 21 flavor. Either way, does Alt-F1 drop you to a command prompt post-login? If so, you may be able to install a new/different DM such as LightDM, then reboot, then see if the problem continues. Wayland support is getting better with each new "underlying" kernel, that linux-surface kernels "extend", so I'd expect more success with 21.10 if you're able to get there from your current build. Wayland is actually more and more the "standard" DE as Xorg approaches its 30th birthday! Or is it a "Compositor", I forget! But try installing something like lightdm-slick-greeter and see if that gets you to your desktop.

**Comment by Amondale** [comment]
You may also try some of the ideas [here](https://blog.beardhatcode.be/2018/03/getting-rid-of-your-dm.html) if the above troubleshooting does in fact point to your DM as the cause of the PSOP.

**Comment by Tom-Finke** [comment]
First of all, thanks for the response.
I totallly forgot to mention the ubuntu version, its 20.04.3 LTS. I'll try upgrading to the newest version and if that doesnt fix it ill look into your other suggestions.

Thank you

**Comment by Tom-Finke** [comment]
I upgraded to Ubuntu 21.10 which didn't fix it. I then installed kubuntu-desktop because i wanted to try kde. Surprisingly, plasma on wayland does work, meaning i dont get the "Purple Screen of Purgatory" as you called it. But waylands performance is very poor, so plasma on wayland is practically unusable for me.
I guess somehow the interplay of GNOME and Wayland is not working, at least not on my particular System.
I will probably try Waylanyd with GNOME when a new release comes out, but for now ill just stick with plasma on x11.
